### PR TITLE
[WIP] Re-viewed quality test interface

### PR DESCRIPTION
**Pull request by rete**
BEGINRELEASENOTES
- QualityTest class
    - Added warning and error limits
    - Added QualityFlag enum as main evaluator of qtest (goes in the report)
- MonitorElement class
    - Added serialisation to json object
- MonitorElementManager class
    - Can read warning/error limits as global or local values for all/single qtest from XML
    - Can write all monitor elements from storage to json
- QReportStorage class
    - Can write all reports to json
- QTest runner
    - Refactored into separate class method to clarify workflow
    - Added possibility to write monitor elements in final QReport
    - Added possibility to compress/beautify json in QReport output file
- Updated quality test unit test

ENDRELEASENOTES

### CODE REVIEW

**Comment by rete** [comment]
Pinging @tcoates3 FYI

File: source/src/MonitorElementManager.cc [hunk 1]
```diff
@@ -230,7 +230,13 @@ namespace dqm4hep {
 
       const bool objectStat(TObject::GetObjectStat());
       TObject::SetObjectStat(false);
-      TObject *pTObject = pTFile->Get(fullName.getPath().c_str());
+      TObject *pTObject = nullptr;
+      // FIXME : bad handling of path in root path
```
Comment (apingault) [inline source/src/MonitorElementManager.cc:234]: Is it still a problem with the way you do it now? I don't see why

**Review by apingault — COMMENTED** [review]
Ok, Just looking at the code I don't see problems. Will have to play with it now ;)

**Comment by rete** [comment]
Qtests are coming

**Comment by rete** [comment]
We can really test it after that

File: source/src/MonitorElementManager.cc (hunk 1 above)
Comment (rete) [reply source/src/MonitorElementManager.cc:234]: Problem is when you use "/" or not to get an object in a root file. At some point I should refactor this in a proper function...

**Pull request merged**